File: Day 6 - Memory Reallocation/Part1.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;


namespace ConsoleApp
{
    class Part1
    {
        static void Main(string[] args)
        {
	    string input = "0	2	7	0";

            List<int> numbers = input.Trim().Split(new char[] { '\t' }, StringSplitOptions.RemoveEmptyEntries).Select(str => int.Parse(str)).ToList();
            HashSet<string> seenStates = new HashSet<string>();
            int steps = 0;
            int length = numbers.Count;

            do {
                steps++;
                int remaining = numbers.Max();
                int indexMax = numbers.IndexOf(remaining);
                int incBy = (int) Math.Round((double) remaining / length);

                for (int i = indexMax + 1; i != indexMax && remaining > 0; i++) {
                    if (i == length) { i = 0; }
                    numbers[i] += incBy;
                    remaining -= incBy;
                }
                numbers[indexMax] = remaining;
            } while (seenStates.Add(string.Join(", ", numbers)));
            

            Debug.WriteLine("steps: " + steps);
	}
    }
}

File: Day 17 - Spinlock/Part1.java
public class Part1 {

    public static void main(String[] args) {
        ArrayList<Integer> values = new ArrayList<>();
        values.add(0);

        int input = 369;
        int position = 0;

        for (int i = 1; i < 2018; i++) {
            position = (position + input) % values.size() + 1;
            values.add(position, i);
        }

        System.out.print(values.get(position) + " -> ");
        System.out.println(values.get(position + 1));
    }

}
